plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/GoogleLoginCommandHandler.java

```diff
@@ -29,26 +34,57 @@
 
 public class GoogleLoginCommandHandler extends AbstractHandler implements IElementUpdater {
 
+  static final IGoogleApiFactory apiFactory = new GoogleApiFactory();
+  static final long LOGGED_IN_CHECK_DELAY = 2000l; // update logged in every 2 seconds
+  static boolean isLoggedIn;
+  Timer loggedInCheckTask = new Timer();
+  
+  public GoogleLoginCommandHandler() {
+    loggedInCheckTask.schedule(new TimerTask() {
+
+      @Override
+      public void run() {
+        checkIsLoggedIn();
```

Comment:
Are we certain this is efficient enough to be running every two seconds?

If we were to change some assumptions (like, displaying the login status in the menu) could we turn this from a polling implementation to a on-demand check?

Reply:
This polling approach was what I tried when noticing that updating the ADC did not reflect in the oauth library (fixed in https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/3716). I will move `checkLoggedIn` to `isEnabled`